### DISCUSSION TITLE
Set sampling rate based on map lookup?

### DISCUSSION THREAD

**Discussion by NeilJed** [vectordotdev/vector#19175] [General]
I've got a scenario where I need to sample different events at different rates depending on their source service. The service name is stored in the field `meta.service`.

I'm wondering if it's possible to set the value of the `rate` parameter via code using a map lookup? My thought is I can allow rates for specific services to be defined and then have a default for anything else.

Not sure if VRL is allowed with the parameter.

**Comment by bruceg** [comment]
There is no way to do as you describe in the `sample` transform at this point. You could, however, accomplish this by using two or more `sample` transforms with different `rate` parameters and then a `route` transform to send different events to the `sample` transform with the desired rate. Finally, the destination sink would then list all of the sample transforms as its `inputs`.

**Comment by NeilJed** [comment]
@bruceg thanks I actually came to the same conclusion and solution myself after a experiementing. I did look at the source code and it seems adding a condition where the result of the VRL needs to be an int could be possible so I've added it to the Ideas thread.
